src/tls/test-connection.c

```diff
@@ -150,6 +157,33 @@ test_read_write (void)
 
   g_assert_cmpint (connection_read (c, WS), ==, RETRY);
 
+  /* test partial writes: send a large block */
+  memset (bigmsg, 42, sizeof (bigmsg));
+  g_assert_cmpint (send (client_fds[1], bigmsg, sizeof (bigmsg), 0), ==, sizeof (bigmsg));
+  g_assert_cmpint (connection_read (c, CLIENT), ==, SUCCESS);
+  g_assert_cmpint (c->buf_client.length, ==, sizeof (bigmsg));
+  /* write to ws should be partial due to our SO_SNDBUF from above */
+  g_assert_cmpint (connection_write (c, CLIENT), ==, PARTIAL);
```

Comment:
This reads wrong to me, like "aren't we writing to WS, not CLIENT"?  I get the whole thing that it's the data *source*, but... meh.  maybe a comment?

Reply:
I adjusted the comment.